controllers/volumereplication_controller.go

```diff
@@ -317,6 +328,50 @@ func (r *VolumeReplicationReconciler) SetupWithManager(mgr ctrl.Manager, cfg *co
 		WithEventFilter(pred).Complete(r)
 }
 
+func (r *VolumeReplicationReconciler) waitForCrds() error {
+
+	logger := r.Log.WithName("checkingDependencies")
+
+	err := r.waitForVolumeReplicationResource(logger, volumeReplicationClass)
+	if err != nil {
+		logger.Error(err, "Failed to watch VolumeReplicationClass")
+		return err
+	}
+
+	err = r.waitForVolumeReplicationResource(logger, volumeReplication)
+	if err != nil {
+		logger.Error(err, "Failed to watch VolumeReplication")
+		return err
+	}
+	return nil
+}
+
+func (r *VolumeReplicationReconciler) waitForVolumeReplicationResource(logger logr.Logger, resourceName string) error {
```

Comment:
don't like the idea of passing logger as function argument.  We should explore other options if possible.

Reply:
why? we do it before in multiple functions? and if you remove it from there, how do you think we should do it differently? creating the logger in the function? I am ok with passing the logger

Reply:
you are right. Its done in other places as well. Ignore my above comment.